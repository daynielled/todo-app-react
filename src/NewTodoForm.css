.form-container {
    margin-bottom: 20px;

    form {
        display: flex;
        align-items: center;


        input {
            flex: 1;
            padding: 8px;
            margin-right: 10px;
        }

        button {
            cursor: pointer;
        }
    }
}