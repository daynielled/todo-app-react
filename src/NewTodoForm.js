import React, { useState } from "react";
import { v4 as uuid } from 'uuid';
import "./NewTodoForm.css";

const NewTodoForm = ({ addNewTodo }) => {
    const [task, setTask] = useState('');
    const inputId = `task-${uuid()}`; //adding an id to eah input
   
    const handleChange = (e) => {
        setTask(e.target.value);
    };


        const handleSubmit = (e) => {
        e.preventDefault();
        addNewTodo({ task, id: uuid() }) 
        setTask('');
    };

    return (
        <div>
            <form  onSubmit={handleSubmit}>
                <label htmlFor={inputId} >Enter Task</label>
                <input
                    id={inputId}
                    name="task"
                    type="text"
                    value={task}
                    onChange={handleChange}
                    placeholder="Enter Task"
                   
                />
                <button>Add Task!</button>
            </form>
        </div>

    );
};

export default NewTodoForm;