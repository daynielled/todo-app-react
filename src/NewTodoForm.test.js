import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

it("renders without crashing", function() {
    render(<NewTodoForm />);
  });
  
  it("matches snapshot", function() {
    const { asFragment } = render(<NewTodoForm />);
    expect(asFragment()).toMatchSnapshot();
  }); 
  
test('renders NewTodoForm component', () => {
  render(<NewTodoForm addNewTodo={() => {}} />);
  const inputElement = screen.getByPlaceholderText('Enter Task');
  expect(inputElement).toBeInTheDocument();
});

test('calls addNewTodo when form is submitted', () => {
  const addNewTodoMock = jest.fn();
  render(<NewTodoForm addNewTodo={addNewTodoMock} />);
  const inputElement = screen.getByPlaceholderText('Enter Task');
  fireEvent.change(inputElement, { target: { value: 'New Task' } });
  const addButton = screen.getByText('Add Task!');
  fireEvent.click(addButton);
  expect(addNewTodoMock).toHaveBeenCalledWith({ id: expect.any(String), task: 'New Task' });
});