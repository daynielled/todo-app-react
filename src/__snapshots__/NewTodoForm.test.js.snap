// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="task-585b5e42-9703-4ef2-adef-2ae70d8a71dd"
      >
        Enter Task
      </label>
      <input
        id="task-585b5e42-9703-4ef2-adef-2ae70d8a71dd"
        name="task"
        placeholder="Enter Task"
        type="text"
        value=""
      />
      <button>
        Add Task!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
